django/tasks/backends/base.py

```diff
@@ -0,0 +1,116 @@
+from abc import ABCMeta, abstractmethod
+from inspect import iscoroutinefunction
+
+from asgiref.sync import sync_to_async
+
+from django.core.checks import messages
+from django.db import connections
+from django.tasks import DEFAULT_QUEUE_NAME
+from django.tasks.exceptions import InvalidTaskError
+from django.tasks.task import MAX_PRIORITY, MIN_PRIORITY, Task
+from django.tasks.utils import is_module_level_function
+from django.utils import timezone
+
+
+class BaseTaskBackend(metaclass=ABCMeta):
+    task_class = Task
+
+    supports_defer = False
+    """Can tasks be enqueued with the run_after attribute"""
+
+    supports_async_task = False
+    """Can coroutines be enqueued"""
+
+    supports_get_result = False
+    """Can results be retrieved after the fact (from **any** thread / process)"""
+
+    def __init__(self, alias, params):
+        self.alias = alias
+        self.queues = set(params.get("QUEUES", [DEFAULT_QUEUE_NAME]))
+        self.enqueue_on_commit = bool(params.get("ENQUEUE_ON_COMMIT", True))
+
+    def _get_enqueue_on_commit_for_task(self, task):
+        """
+        Determine the correct `enqueue_on_commit` setting to use for a given task.
+
+        If the task defines it, use that, otherwise, fall back to the backend.
+        """
+        # If this project doesn't use a database, there's nothing to commit to
+        if not connections.settings:
+            return False
```

Comment:
This doesn't have test coverage and given that we have a system check to error when `enqueue_on_commit` is True, should we not error out when a task has this set to True?